src/geometry/to_bevy_mesh.rs

```diff
@@ -0,0 +1,230 @@
+                bevy::render::mesh::PrimitiveTopology::TriangleList,
+                RenderAssetUsages::default(),
+            )
+            .with_inserted_indices(Indices::U32(indices.iter().cloned().flatten().collect()));
+            let mesh = mesh.with_inserted_attribute(Mesh::ATTRIBUTE_POSITION, vertices);
+
+            mesh.into()
+        }
+        rapier::prelude::TypedShape::Polyline(polyline) => {
+            // FIXME: Polyline shape not implemented yet, how to represent it ? BoxedPolyline3d is only a primitive.
+            return None;
+        }
+        rapier::prelude::TypedShape::HalfSpace(half_space) => {
+            // FIXME: HalfSpace shape not implemented yet, how to represent it ? its infinite property makes it difficult.
+            return None;
+        }
+        rapier::prelude::TypedShape::HeightField(height_field) => {
+            #[cfg(feature = "dim2")]
+            // FIXME: "HeightField for 2d not implemented yet, how to represent it ? its effectively a line.
+            return None;
+            #[cfg(feature = "dim3")]
+            {
+                // FIXME: we could use TriMesh::From(height_field), but that would clone, we should fix that in parry.
+                let (vtx, idx) = height_field.to_trimesh();
+                let tri_mesh = TriMesh::new(vtx, idx).unwrap();
+
+                // From Trimesh:
+                let vertices = tri_mesh.vertices();
+                let vertices: Vec<_> = vertices.iter().map(|pos| [pos.x, pos.y, 0.0]).collect();
+                let indices = tri_mesh.indices();
+                let mesh = Mesh::new(
+                    bevy::render::mesh::PrimitiveTopology::TriangleList,
+                    RenderAssetUsages::default(),
+                )
+                .with_inserted_indices(Indices::U32(indices.iter().cloned().flatten().collect()));
+                let mesh = mesh.with_inserted_attribute(Mesh::ATTRIBUTE_POSITION, vertices);
+
+                mesh.into()
+            }
+        }
+        rapier::prelude::TypedShape::Compound(compound) => {
+            let mut vertices = Vec::new();
+            let mut indices = Vec::new();
+            for shape in compound.shapes() {
+                let typed_shape = shape.1.as_typed_shape();
+                let mesh = typed_shape_to_mesh(&typed_shape)?;
+
+                assert!(mesh.primitive_topology() == bevy::render::mesh::PrimitiveTopology::TriangleList,
+                "Compound shape mesh conversion does not support shapes not converting to PrimitiveTopology::TriangleList.");
+
+                vertices.append(
+                    &mut mesh
+                        .attribute(Mesh::ATTRIBUTE_POSITION.id)
+                        .unwrap()
+                        .as_float3()
+                        .unwrap()
+                        .iter()
+                        .cloned()
+                        .flatten()
+                        .collect::<Vec<_>>(),
+                );
```

Comment:
is there a bevy builtin to merge meshes ?

Reply:
Yes there is, but it'd be great if it'd be faillible ( https://github.com/bevyengine/bevy/pull/17475 )